subsys/sd/mmc.c

```diff
@@ -34,14 +50,491 @@ inline int mmc_ioctl(struct sd_card *card, uint8_t cmd, void *buf)
 }
 /* -------------------------------- */
 
+/* Send CMD1 */
+static int mmc_send_op_cond(struct sd_card *card, int ocr);
+
+/* Send CMD3 */
+static int mmc_set_rca(struct sd_card *card);
+
+/* Send CMD9 */
+static int mmc_read_csd(struct sd_card *card, struct sd_csd *card_csd);
+static inline void mmc_decode_csd(
+		struct sd_csd *csd, uint32_t *raw_csd);
+
+/* Send CMD8 */
+static int mmc_read_ext_csd(struct sd_card *card, struct mmc_ext_csd *card_ext_csd);
+static inline void mmc_decode_ext_csd(
+		struct mmc_ext_csd *ext_csd, uint8_t *raw);
+
+static inline int mmc_set_max_freq(
+		struct sd_card *card, struct sd_csd *card_csd);
+static int mmc_set_timing(struct sd_card *card);
+
+/* CMD6 switch bus width*/
+static int mmc_set_bus_width(struct sd_card *card);
+/* CMD19 and CMD14 */
+static int mmc_bus_test(struct sd_card *card);
+
 /*
  * Initialize MMC card for use with subsystem
  */
 int mmc_card_init(struct sd_card *card)
 {
 	int ret;
+	uint32_t ocr_arg = 0U;
```

Comment:
Nit: you may want to check if the host controller uses SPI mode at this point, and return an error if it does (so SPI mode controllers cannot attempt to probe eMMC cards)